### PR TITLE
Prevent GPS injection from choking the GPS driver

### PR DESCRIPTION
**Pull request by potaito**
Solves https://github.com/PX4/Firmware/issues/11716

FYI @tdnet12434 

**Describe problem solved by the proposed pull request**
Sending a continuous stream of injection data can cause the GPS driver to get stuck indefinitely in the handling loop.

**Test data / coverage**
We tested with a proprietary module able to spam GPS injection data for a few seconds.
@tdnet12434 came to the same conclusion individually: https://github.com/PX4/Firmware/issues/11716#issuecomment-502011464

**Describe your preferred solution**
This proposed solution is a bit hacky, since it uses a hard-coded integer for the maximum number of consecutive injection message to handle. We are limiting to 6 consecutive messages, since a GPS injection data set can be split and sent as multiple messages. Reading only a single message per loop might yield incomplete injection data. According to our knowledge, no injection data set is larger than 3 messages.

**Describe possible alternatives**
Any solution not involving a magic number.

### CODE REVIEW

File: src/drivers/gps/gps.cpp [hunk 1]
```diff
@@ -416,7 +416,10 @@ void GPS::handleInjectDataTopic()
 {
 	bool updated = false;
 
+	const size_t max_num_injections = 6;
```
Comment (bkueng) [inline src/drivers/gps/gps.cpp]: Magic numbers :-) Can you add a comment why it's 6?

Comment (potaito) [reply src/drivers/gps/gps.cpp]: Fair point :+1:

Comment (potaito) [reply src/drivers/gps/gps.cpp]: Problem is I don't remember exactly the argument for 6. I think the injection data that belongs together is split into multiple messages for multiple sattelite networks. So Galileo, Baidu etc.
Let me ask someone who knows for sure.

**Comment by potaito** [comment]
My apologies. I left for vacation without completing this PR. Thanks @bkueng and @dagar for finishing it. Much appreciated :+1:

**Pull request merged**